src/application/machines/reconnectMachine.tsx

```diff
@@ -0,0 +1,52 @@
+import { setup } from "xstate";
+import { BannerAlert } from "../components/BannerAlert";
+
+export type ReconnectMachineEvents = { type: "reconnect.retry" };
+
+export const reconnectMachine = setup({
+  types: {} as {
+    events: ReconnectMachineEvents;
+  },
+  delays: {
+    connectTimeout: 10_000,
+  },
+  actions: {
+    notifyWaitingForConnection: () => {
+      BannerAlert.show({
+        type: "loading",
+        content: "Waiting for client to connect...",
+      });
+    },
+    notifyNotFound: () => {
+      BannerAlert.show({
+        type: "error",
+        content: "Client not found",
+      });
+    },
+  },
+}).createMachine({
+  initial: "disconnected",
+  states: {
+    disconnected: {
+      after: {
+        connectTimeout: {
+          target: "notFound",
+        },
+      },
+    },
+    retrying: {
+      entry: ["notifyWaitingForConnection"],
+      after: {
+        connectTimeout: {
+          target: "notFound",
+        },
+      },
+    },
+    notFound: {
+      on: {
+        "reconnect.retry": "retrying",
+      },
+      entry: ["notifyNotFound"],
+    },
```

Comment:
Previously, this also had a `timedout` state, but we never triggered it from anywhere.

Reply:
I'm glad you got rid of it. It was a relic from the days before we had the client send a register event. Devtools used to ping the content scripts, so the `timedout` event was used if devtools couldn't communicate with the content scripts anymore, but that is largely a thing of the past.